geonode/layers/utils.py

```diff
@@ -788,6 +790,19 @@ def create_thumbnail(instance, thumbnail_remote_url, thumbnail_create_url=None,
 
         if valid_x and valid_y:
             logger.debug('Valid bbox. Will create Remote Thumbnail Link.')
+
+            if check_ogc_backend(qgis_server.BACKEND_PACKAGE):
+                # Only for QGIS Server
+                # We don't want to add the internal QGIS IP.
+                # We need the public URL of Geonode.
+                public_url = settings.SITEURL
+                if public_url.endswith('/'):
+                    # reverse will give also a slash. We need only one.
+                    public_url = public_url[0:-1]
+                public_url += reverse('qgis_server:request')
+                thumbnail_remote_url = thumbnail_remote_url.replace(
+                    QGIS_SERVER_CONFIG['qgis_server_url'], public_url)
```

Comment:
Yes, what I meant is QGIS Server url (the one that is not allowed to be accessed from public location). We don't have this kind of problem because GeoServer has its own public url. But for QGIS Server backend, we had to refactor some tests because the test suite will not find its own address. The url is different inside and outside. So I think it's better if you remove this and I need to fix it on a different PR.